File: bin/wlanstack/src/logger.rs
// Copyright 2018 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#![deny(warnings)]

use log::{self, LogLevel, LogMetadata, LogRecord};

const LOG_LEVEL: LogLevel = LogLevel::Debug;

pub struct Logger;

fn short_log_level(level: &LogLevel) -> &'static str {
    match *level {
        LogLevel::Error => "E",
        LogLevel::Warn => "W",
        LogLevel::Info => "I",
        LogLevel::Debug => "D",
        LogLevel::Trace => "T",
    }
}

impl log::Log for Logger {
    fn enabled(&self, metadata: &LogMetadata) -> bool {
        metadata.level() <= LOG_LEVEL
    }
    fn log(&self, record: &LogRecord) {
        if self.enabled(record.metadata()) {
            println!(
                "{} [{}]: {}",
                record.target(),
                short_log_level(&record.level()),
                record.args()
            );
        }
    }
}

File: src/lib/fidl_codec/logger.cc
// Copyright 2020 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "src/lib/fidl_codec/logger.h"

namespace fidl_codec::logger::internal {

thread_local std::ostream* log_stream_tls = nullptr;

}  // namespace fidl_codec::logger::internal
